pkg/util/tar_utils.go

```diff
@@ -68,6 +68,13 @@ func unpackTar(tr *tar.Reader, path string, whitelist []string) error {
 		// if its a dir and it doesn't exist create it
 		case tar.TypeDir:
 			if _, err := os.Stat(target); os.IsNotExist(err) {
+				if mode.Perm()&(1<<(uint(7))) == 0 {
+					logrus.Debugf("Write permission bit not set on %s by default; setting manually", target)
+					originalMode := mode
+					mode = mode | (1 << uint(7))
+					defer os.Chmod(target, originalMode)
```

Comment:
Chmod can return an Error, might be worth not deferring it and handling the error

Reply:
good call